### PR TITLE
Vuln v2 views apis test

### PR DESCRIPTION
**Pull request by avrahams**
## **Type**
enhancement, tests

___

## **Description**
- Introduced new API endpoints and handling methods for vulnerability V2 views.
- Added a comprehensive set of tests and expected results for vulnerability V2 views, including workloads, images, components, and CVE details.
- Enhanced test utilities with JSON comparison capabilities for verifying expected vs actual results.
- Updated documentation and test mappings to include the new vulnerability V2 views test case.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests
</strong></td><td><details><summary>6 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>vuln_scan_tests.py</strong><dd><code>Add New Vulnerability V2 Views Test Case</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

configurations/system/tests_cases/vuln_scan_tests.py
<li>Added a new test case method <code>vuln_v2_views</code> for vulnerability V2 views <br>testing.<br>

</details>
    

  </td>
  <td><a href="https://github.com/armosec/system-tests/pull/275/files#diff-5a30ae714bdf65ead73819f4cdc8869036805e89f3fbc691cb5d70dd92a1fcde">+19/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>vuln_scan.py</strong><dd><code>Implement Vulnerability V2 Views Testing Logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests_scripts/helm/vuln_scan.py
<li>Added a new class <code>VulnerabilityV2Views</code> for testing vulnerability V2 <br>views.<br> <li> Implemented methods for setup, applying resources, verifying pods, and <br>comparing expected results with actual API responses.<br>

</details>
    

  </td>
  <td><a href="https://github.com/armosec/system-tests/pull/275/files#diff-288de96b50f4d3d32f61bc4d91633848bd020741853e6416e73f7c9a1ee415e0">+128/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>cve_details.json</strong><dd><code>Add Expected CVE Details JSON for Vulnerability V2 Views</code>&nbsp; </dd></summary>
<hr>

configurations/expected-result/V2_VIEWS/cve_details.json
<li>Added expected JSON result for CVE details in vulnerability V2 views.<br>

</details>
    

  </td>
  <td><a href="https://github.com/armosec/system-tests/pull/275/files#diff-eeeaf6a93fa7c06437888f6d9515ae238441c6be1c0ebb0fdc32796cc03cdb46">+132/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>image_details.json</strong><dd><code>Add Expected Image Details JSON for Vulnerability V2 Views</code></dd></summary>
<hr>

configurations/expected-result/V2_VIEWS/image_details.json
<li>Added expected JSON result for image details in vulnerability V2 <br>views.<br>

</details>
    

  </td>
  <td><a href="https://github.com/armosec/system-tests/pull/275/files#diff-c7794d447cde428a689a811888f33bfc34c0ed3d618225fd65184c7f02e9dd66">+67/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>wl_details.json</strong><dd><code>Add Expected Workload Details JSON for Vulnerability V2 Views</code></dd></summary>
<hr>

configurations/expected-result/V2_VIEWS/wl_details.json
<li>Added expected JSON result for workload details in vulnerability V2 <br>views.<br>

</details>
    

  </td>
  <td><a href="https://github.com/armosec/system-tests/pull/275/files#diff-1972dee7a7c5f85c1af4dadd300736dc295b4816636d7a8b040c6958c08ae42d">+68/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>wl_filtered.json</strong><dd><code>Add Expected Filtered Workload JSON for Vulnerability V2 Views</code></dd></summary>
<hr>

configurations/expected-result/V2_VIEWS/wl_filtered.json
<li>Added expected JSON result for filtered workload in vulnerability V2 <br>views.<br>

</details>
    

  </td>
  <td><a href="https://github.com/armosec/system-tests/pull/275/files#diff-b46d2578828c8a0c2de4c118a1e1678f41f2dda6c5f371314465ec0ba376ee4e">+28/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Enhancement
</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>backend_api.py</strong><dd><code>Introduce Vulnerability V2 API Endpoints and Handling Methods</code></dd></summary>
<hr>

infrastructure/backend_api.py
<li>Introduced new API endpoints for vulnerability V2 views.<br> <li> Added methods to handle requests for vulnerability V2 workloads, <br>images, components, and details.<br>

</details>
    

  </td>
  <td><a href="https://github.com/armosec/system-tests/pull/275/files#diff-a386f72e107e19ee1faa1c9946d320d17d3ce39a24ea1cacee998ae67a7db245">+57/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>systests_utilities.py</strong><dd><code>Enhance Test Utilities with JSON Comparison Capabilities</code>&nbsp; </dd></summary>
<hr>

systest_utils/systests_utilities.py
<li>Imported <code>DeepDiff</code> for JSON comparison.<br> <li> Added utility methods for handling expected JSON files and comparing <br>JSONs.<br>

</details>
    

  </td>
  <td><a href="https://github.com/armosec/system-tests/pull/275/files#diff-3fd0cc2a3f85b63759f40136f99d484e69d144573264ee843f3202d4f17394da">+25/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Documentation
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>readme.md</strong><dd><code>Document New Vulnerability V2 Views Test Case</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

readme.md
<li>Documented the new <code>vuln_v2_views</code> test case in the test cases list.<br>

</details>
    

  </td>
  <td><a href="https://github.com/armosec/system-tests/pull/275/files#diff-5a831ea67cf5cf8703b0de46901ab25bd191f56b320053be9332d9a3b0d01d15">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>system_test_mapping.json</strong><dd><code>Add Test Mapping for Vulnerability V2 Views Test Case</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

system_test_mapping.json
- Added mapping for the new `vuln_v2_views` test case.

</details>
    

  </td>
  <td><a href="https://github.com/armosec/system-tests/pull/275/files#diff-7ac9a8e9fb7431bc23f91250ada3220ba55bdcb91d6a30b72ee1ab242a88e78d">+12/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: system_test_mapping.json [hunk 1]
```diff
@@ -289,6 +289,18 @@
...
     },
+    "vuln_v2_views": {
+        "target": [
+            "In cluster",
+            "Backend"
+        ],
+        "target_repositories": [
+            "cadashboardbe",
+            "careportsreceiver"
+        ],
```
Comment (kooomix) [inline system_test_mapping.json:300]: @avrahams we should also have the ingesters repository target, no?

**Review by kooomix — CHANGES_REQUESTED** [review]

File: system_test_mapping.json (hunk 1 above)
Comment (avrahams) [reply system_test_mapping.json:300]: @kooomix 
I am not sure, the new apis do not add any new data in PG
The new API is all about aggregation on the existing data with read SQL queries 
Actually we may want it only for the dashboard - WDYT ?

File: tests_scripts/helm/vuln_scan.py [hunk 2]
```diff
@@ -862,3 +862,129 @@ def check_wrong_auth(self):
...
+            "kind": wl_summary["kind"],
+            "workload": wl_summary["name"]
+            }]}
+        components = self.backend.get_vuln_v2_components(body=body)
+        assert len(components) == 9, 'expect 9 components found for httpd-proxy'
+      
+        Logger.logger.info('7. get workloads images and compare with expected')
+        image = self.backend.get_vuln_v2_images(body=body, expected_results=wl_summary["imagesCount"])
+        image = image[0]
+        image_excluded_paths = {"root['lastScanTime']", "root['customerGUID']","root['cisaKevInfo']"}
```
Comment (refaelm92) [inline tests_scripts/helm/vuln_scan.py:949]: cisaKevInfo on images?

File: tests_scripts/helm/vuln_scan.py [hunk 3]
```diff
@@ -862,3 +862,129 @@ def check_wrong_auth(self):
...
+        Logger.logger.info('7. get exploitable CVE details and compare with expected')        
+        body['innerFilters'][0]['severity'] = ""
+        body['innerFilters'][0]['exploitable'] = "Known Exploited,High Likelihood"
+        cve = self.backend.get_vulns_v2(body=body, expected_results=1)
+        cve = cve[0]
+        body['innerFilters'][0]['name'] = cve["name"]
+        body['innerFilters'][0]['componentInfo.name'] = cve["componentInfo"]["name"]
+        body['innerFilters'][0]['componentInfo.version'] = cve["componentInfo"]["version"]
+        cve = self.backend.get_vuln_v2_details(body=body)
+        cve_excluded_paths = {"root['links']", "root['epssInfo']","root['cisaKevInfo']","root['componentInfo']['pathsInfo'][0]['workloadHash']","root['componentInfo']['pathsInfo'][0]['clusterName']"}
```
Comment (refaelm92) [inline tests_scripts/helm/vuln_scan.py:986]: lets exclude also cvssInfo.baseScore

File: system_test_mapping.json (hunk 1 above)
Comment (avrahams) [reply system_test_mapping.json:300]: So here there is no change in the way we collect or store the data
These are all views of the existing data that is tested on other vulan scan test 
So setting it as dashboard only test

**Review by kooomix — APPROVED** [review]

**Pull request merged**